src/Robo/Plugin/Traits/SitesConfigTrait.php

```diff
@@ -78,12 +78,17 @@ protected function getSiteConfig(string $siteName = 'default'): array
      *   The site configuration key to load.
      * @param string $siteName
      *   The site name.
+     * @param bool $required
+     *   Whether the config item is expected to always be present.
      */
-    public function getSiteConfigItem(string $key, string $siteName = 'default'): mixed
+    public function getSiteConfigItem(string $key, string $siteName = 'default', bool $required = true): mixed
     {
         $siteConfig = $this->getSiteConfig(siteName: $siteName);
         if (!isset($siteConfig[$key])) {
-            throw new TaskException($this, "Key $key not found for '$siteName' in $this->sitesConfigFile.");
+            if ($required) {
+                throw new TaskException($this, "Key $key not found for '$siteName' in $this->sitesConfigFile.");
+            }
+            return null;
```

Comment:
This seems overly complicated. If we are offering contextual ability to ignore a missing config, then why not return an empty config when missing and just test for it/decide what to do, in the calling code? 

Alternatively, it seems if there really are properties that are "generally required" (ie, things will break without them), then that should be decided at the application level and not be able to be overridden by the caller. Hope this makes sense. Now that I read it, I'm not so sure 😆

Reply:
I'm not 100% sure I follow this, so consider me on the fence.

Reply:
> This seems overly complicated. If we are offering contextual ability to ignore a missing config, then why not return an empty config when missing and just test for it/decide what to do, in the calling code?

I understand what you are driving at. It started this way, but then I had a bunch of code calling this method, and in most cases that code was all starting to look very similar with conditional checks and exceptions. My thought was I could simplify all of those calls and just do the check/exception here in `getSiteConfigItem`. I see the argument in both directions, but personally, I just got sick of writing the checks and exception code every time I wanted to call into site config. 

Another option would be to create two helper methods `getSiteConfigItemRequired` and `getSiteConfigItemOptional`, but that is just a different flavor of what we have here.

Thoughts?

> Alternatively, it seems if there really are properties that are "generally required" (ie, things will break without them), then that should be decided at the application level and not be able to be overridden by the caller.

Which application are we talking about here?

Reply:
Following up here, @markdorison and I discussed defining a list of required configs and using that list in validation here. ie, if the config requested is empty and is in the required list, throw the exception, don't throw if not.

Reply:
I created https://github.com/ChromaticHQ/usher/issues/169 for this topic!